private/buf/bufwire/image_config_reader.go

```diff
@@ -174,6 +176,12 @@ func (i *imageConfigReader) getSourceOrModuleImageConfigs(
 			return nil, nil, err
 		}
 		if imageConfig != nil {
+			if importsErr := i.validateSourceImports(
+				moduleConfig.Module().DirectDependencies(),
+				imageConfig.Image(),
+			); importsErr != nil {
+				return nil, nil, fmt.Errorf("source imports: %w", importsErr)
```

Comment:
Locally this outputs something like this:

```
$ bufltp build -v --debug
DEBUG	bufwire	module	{"module_direct_dependencies": {"buf.build/jfigueroa/students":{}}}
DEBUG	bufwire	image_files	{"all_with_imports": {"buf.build/jfigueroa/people":{"people/v1/other_people.proto":null,"people/v1/people.proto":null},"buf.build/jfigueroa/students":{"students/v1/students.proto":["people/v1/people.proto","people/v1/other_people.proto"]},"source":{"school/v1/school.proto":["students/v1/students.proto","people/v1/people.proto","people/v1/other_people.proto"]}}, "from_direct_dependencies": {"students/v1/students.proto":"buf.build/jfigueroa/students"}, "from_transitive_dependencies": {"people/v1/other_people.proto":"buf.build/jfigueroa/people","people/v1/people.proto":"buf.build/jfigueroa/people"}}
DEBUG	get_ref	{"duration": "83.5µs"}
DEBUG	get_config	{"duration": "578.042µs"}
DEBUG	get_config	{"duration": "33.333µs"}
DEBUG	get_module_config	{"duration": "1.72675ms"}
DEBUG	get_config	{"duration": "30.125µs"}
DEBUG	get_config	{"duration": "26.417µs"}
DEBUG	get_image	{"duration": "23.417µs"}
DEBUG	build	{"duration": "5.690125ms"}
DEBUG	build_module	{"duration": "5.692917ms"}
DEBUG	command	{"duration": "10.651209ms"}
Failure: source imports: source proto file "school/v1/school.proto" imports "people/v1/people.proto", not found in your direct dependencies, but found in transitive dependency "buf.build/jfigueroa/people", please declare that one as explicit dependency in your buf.yaml file; source proto file "school/v1/school.proto" imports "people/v1/other_people.proto", not found in your direct dependencies, but found in transitive dependency "buf.build/jfigueroa/people", please declare that one as explicit dependency in your buf.yaml file
```

Which is a pretty nice DX imo

Reply:
Wow, seeing a lot of broken tests with this message, will check them out.

Reply:
There's no need for this `source imports: ` qualifier, its extraneous, remove wrapping of error message